client/my-sites/upgrades/domain-management/dns/dns-add-new.jsx

```diff
@@ -166,7 +169,7 @@ const DnsAddNew = React.createClass( {
 
 				<FormFooter>
 					<FormButton
-						disabled={ formState.isSubmitButtonDisabled( this.state.fields ) || this.props.isSubmittingForm }
+						disabled={ this.state.show ? isSubmitDisabled : false }
```

Comment:
Surprisingly - yes. The way the code is written right now, the same button handles toggling the form _and_ submitting it. So the button is always visible even if the form is not.
I know that it's bad that it's not obvious from the code more (and I hate the generic `show` variable name), but I didn't want to introduce too many changes, so I left it as-is for now.